kubectl-plugin/pkg/cmd/ray.go

```diff
@@ -29,14 +31,19 @@ func NewRayCommand(streams genericiooptions.IOStreams) *cobra.Command {
 		},
 	}
 
-	cmd.AddCommand(get.NewGetCommand(streams))
-	cmd.AddCommand(session.NewSessionCommand(streams))
-	cmd.AddCommand(log.NewClusterLogCommand(streams))
-	cmd.AddCommand(job.NewJobCommand(streams))
-	cmd.AddCommand(version.NewVersionCommand(streams))
-	cmd.AddCommand(create.NewCreateCommand(streams))
-	cmd.AddCommand(kubectlraydelete.NewDeleteCommand(streams))
-	cmd.AddCommand(scale.NewScaleCommand(streams))
+	configFlags := genericclioptions.NewConfigFlags(true)
+	configFlags.AddFlags(cmd.PersistentFlags())
+
+	cmdFactory := cmdutil.NewFactory(configFlags)
+
+	cmd.AddCommand(get.NewGetCommand(cmdFactory, streams))
+	cmd.AddCommand(session.NewSessionCommand(cmdFactory, streams))
```

Comment:
We need to create a shared `cmdFactory` at the root here and pass to all sub-commands so they can access processed global flag values like `--context` later.